chalk-engine/src/forest.rs

```diff
@@ -54,89 +53,6 @@ impl<I: Interner, C: Context<I>> Forest<I, C> {
             _context: std::marker::PhantomData::<C>,
         }
     }
-
-    /// Solves a given goal, producing the solution. This will do only
-    /// as much work towards `goal` as it has to (and that works is
-    /// cached for future attempts).
-    pub fn solve(
```

Comment:
As sort of a drive-by change, I moved these two functions `solve` and `solve_multiple` out of `Forest` (and `chalk-engine`) and into `chalk_solve::solve`. For `solve`, it honestly just makes more sense, since this is the *only* place solution is needed. For `solve_multiple`, there's a little work done here, but really I think these are implementation details akin to `aggregate`.